JsonSchema/ErrorMessages.cs

```diff
@@ -51,27 +45,82 @@ private static string Get(CultureInfo? culture = null, [CallerMemberName] string
-    [UnconditionalSuppressMessage("AOT", "IL3050:Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code", Justification = "We guarantee that the SerializerOptions covers all the types we need for AOT scenarios.")]
-    public static string ReplaceTokens(this string message, params (string token, object? value)[] parameters)
+	/// <summary>
+	/// Replaces tokens in the form of `[[token]]` with a specified value, serialized as JSON.
+	/// </summary>
+	/// <param name="message">The message template.</param>
+	/// <param name="name">The token name (without brackets)</param>
+	/// <param name="value">The value.</param>
+	/// <returns>The detokenized string.</returns>
+	[RequiresDynamicCode("Calls JsonSerializer.Serialize and might require reflection for token values serialization. Prefer ReplaceToken that takes TypeInfo parameters for AOT scenarios.")]
+	[RequiresUnreferencedCode("Calls JsonSerializer.Serialize and might require reflection for token values serialization. Prefer ReplaceToken that takes TypeInfo parameters for AOT scenarios.")]
+	public static string ReplaceToken(this string message, string name, object? value)
     {
-        var current = message;
-        var values = new object[parameters.Length];
-        for (var i = 0; i < parameters.Length; i++)
-        {
-            var parameter = parameters[i];
-            values[i] = JsonSerializer.Serialize(parameter.value, _serializerOptions);
-            current = current.Replace($"[[{parameter.token}]]", $"{{{i}}}");
-        }
+		var stringValue = JsonSerializer.Serialize(value);
+		return message.Replace($"[[{name}]]", stringValue);
+	}
 
-        return string.Format(current, values);
-    }
-}
\ No newline at end of file
+	/// <summary>
+	/// Replaces tokens in the form of `[[token]]` with a specified value, serialized as JSON.
+	/// </summary>
+	/// <param name="message">The message template.</param>
+	/// <param name="name">The token name (without brackets)</param>
+	/// <param name="value">The value.</param>
+	/// <param name="typeInfo">A JsonTypeInfo that matches <paramref name="value"/>.</param>
+	/// <returns>The detokenized string.</returns>
+	public static string ReplaceToken<T>(this string message, string name, T? value, JsonTypeInfo<T> typeInfo)
+	{
+		var stringValue = JsonSerializer.Serialize(value, typeInfo!);
+		return message.Replace($"[[{name}]]", stringValue);
+	}
+
+	/// <summary>
+	/// Replaces tokens in the form of `[[token]]` with a specified value, serialized as JSON.
+	/// </summary>
+	/// <param name="message">The message template.</param>
+	/// <param name="name">The token name (without brackets)</param>
+	/// <param name="value">The value.</param>
+	/// <returns>The detokenized string.</returns>
+	public static string ReplaceToken(this string message, string name, int value) => 
```

Comment:
Now that you've done these, I suppose we'd keep them, but I would like one for `JsonNode` to allow users to pre-serialize values.

Reply:
Having the strong overloads is a life saver, IMO. I just added a JsonNode version.

Reply:
I imagine it gives a sufficient example to form a pattern for others who want to make their own strongly typed ones as well.